frontend/ws3dash.Rmd

```diff
@@ -0,0 +1,170 @@
+---
+title: "Central Florida Event Siting Tool Output"
+output:
+  flexdashboard::flex_dashboard
+---
+
+```{r setup, include=FALSE, message=FALSE, warning=FALSE}
+library(flexdashboard)
+library(sf)
+library(mapview)
+library(tidyverse)
+library(DT)
+library(here)
+```
+
+```{r setup-parameters}
+# Set up main mapping parameters here
+
+# how many points to display on the map
+MAP_SITES_TO_DISPLAY <- 100
+# how many rows to display on the table
+TABLE_SITES_TO_DISPLAY <- 25
+# how many decimals to round the event score to
+EVENT_SCORE_NUM_DECIMALS <- 2
+```
+
+```{r setup-functions}
+# Mapping and table display functions defined below
+# These functions are used for every county
+
+show_map <- function(df) {
+  #' Takes a dataframe-like object, filters points to disply and renders a map
+  #'
+  #' @param df The data to plot on a map
+
+  # Filter sites to display
+  county_data <- df %>%
+    slice_head(n = MAP_SITES_TO_DISPLAY)
+
+  # Render map
+  mapview(county_data, zcol = "event_score", layer.name = "Event Score")
+}
+
+
+show_table <- function(df) {
+  #' Takes a dataframe-like object, formats contents and renders a table
+  #'
+  #' @param df The data to display as a table
+
+  # Select table attributes to display, round long decimals and sort rows
+  county_table <- df %>%
+    dplyr::select(name, type, event_score, address, website, phone_number) %>%
+    dplyr::mutate(event_score = round(event_score, EVENT_SCORE_NUM_DECIMALS)) %>%
+    arrange(desc(event_score))
+
+  # Render table
+  DT::datatable(county_table,
+                rownames = FALSE,
+                options = list(pageLength = TABLE_SITES_TO_DISPLAY , scrollX = TRUE),
+                class = 'white-space: nowrap')
+}
```

Comment:
Similarly I organized all the table-formatting code into one function. Notably all the rounding and arranging is done only here since it's not needed in the map.